internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -46,10 +47,11 @@ func resourceCdnFrontDoorSecurityPolicy() *pluginsdk.Resource {
 			},
 
 			"cdn_frontdoor_profile_id": {
-				Type:         pluginsdk.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validate.FrontDoorProfileID,
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateFunc:     validate.FrontDoorProfileID,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
We only use DiffSuppressFunc's in the schema if the casing of user specified segments are not preserved by the API. From the issue description it seems that the RP segment `microsoft.network` is the culprit here, which means we should be insensitively parsing this in the read instead of suppressing the case here.

Reply:
> We only use DiffSuppressFunc's in the schema if the casing of user specified segments are not preserved by the API. From the issue description it seems that the RP segment `microsoft.network` is the culprit here, which means we should be insensitively parsing this in the read instead of suppressing the case here.

Sorry, this is out of my knowledge of azurerm provider internals.